apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx

```diff
@@ -59,14 +60,46 @@ const QuizForm: React.FC<Properties> = ({ onNext }: Properties) => {
 		void dispatch(quizActions.previousQuestion());
 	}, [dispatch]);
 
-	const handleNextStep = useCallback(() => {
-		if (isLast) {
-			return;
-		}
+	const checkIfValueIsUndefined = useCallback(
+		(formData: FormToSave, keysToCheck: string[]): boolean => {
+			return keysToCheck.some((key) => formData[key] === undefined);
+		},
+		[],
+	);
+
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
+
+	const handleNextStep = useCallback(
+		(data: QuizFormValues) => {
+			if (!category) {
+				return;
+			}
+
+			const ids = category.map(
+				(categoryItem) => `answer${categoryItem.id.toString()}`,
```

Comment:
Try to run app and check how it works. Your checkIfValueIsUndefined function always returns true, because you provided wrong ids. So user can't go next even after answering all questions.
```suggestion
				(categoryItem) => `question${categoryItem.id.toString()}`,
```